cli/lsp/tsc.rs

```diff
@@ -45,12 +50,30 @@ type Request = (
   oneshot::Sender<Result<Value, AnyError>>,
 );
 
+type DiagnosticsRequest = (
+  HashSet<ModuleSpecifier>,
+  StateSnapshot,
+  oneshot::Sender<Result<Option<Value>, AnyError>>,
+);
+
 #[derive(Clone, Debug)]
-pub struct TsServer(mpsc::UnboundedSender<Request>);
+pub struct TsServer {
+  requests_queue: mpsc::UnboundedSender<Request>,
+  diagnostics_notifier: mpsc::Sender<()>,
+  diagnostics_request: Arc<Mutex<Option<DiagnosticsRequest>>>,
+}
 
 impl TsServer {
   pub fn new() -> Self {
-    let (tx, mut rx) = mpsc::unbounded_channel::<Request>();
+    let (requests_queue_tx, mut requests_queue_rx) =
+      mpsc::unbounded_channel::<Request>();
+    let (diagnostics_notifier_tx, mut diagnostics_notifier_rx) =
+      mpsc::channel::<()>(1);
+    let diagnostics_request: Arc<Mutex<Option<DiagnosticsRequest>>> =
+      Default::default();
+
+    let diagnostics_request_ = diagnostics_request.clone();
```

Comment:
```suggestion
    let moved_diagnostics_request = diagnostics_request.clone();
```

Or anything than a trailing `_` as I personally think it is just ambiguous what it represented. `diagnostics_request_clone` or whatever...